src/main/java/org/folio/service/transactions/batch/BatchTransactionService.java

```diff
@@ -0,0 +1,230 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.transactions.BatchTransactionDAO;
+import org.folio.okapi.common.OkapiToken;
+import org.folio.okapi.common.XOkapiHeaders;
+import org.folio.rest.core.model.RequestContext;
+import org.folio.rest.exception.HttpException;
+import org.folio.rest.jaxrs.model.Batch;
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.Metadata;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.model.Transaction.TransactionType;
+import org.folio.rest.jaxrs.model.TransactionPatch;
+import org.folio.rest.persist.DBClient;
+import org.folio.rest.persist.DBClientFactory;
+import org.folio.rest.persist.DBConn;
+import org.folio.service.budget.BudgetService;
+import org.folio.service.fund.FundService;
+import org.folio.service.ledger.LedgerService;
+
+import java.util.Date;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.ALLOCATION;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.CREDIT;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.ENCUMBRANCE;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.PAYMENT;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.PENDING_PAYMENT;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.TRANSFER;
+
+public class BatchTransactionService {
+  private static final Logger logger = LogManager.getLogger();
+  private static final List<TransactionType> transactionTypesInOrder = List.of(ALLOCATION, TRANSFER, PAYMENT,
+    PENDING_PAYMENT, ENCUMBRANCE);
+  private final DBClientFactory dbClientFactory;
+  private final BatchTransactionDAO transactionDAO;
+  private final FundService fundService;
+  private final BudgetService budgetService;
+  private final LedgerService ledgerService;
+  private final Map<TransactionType, BatchTransactionServiceInterface> serviceMap;
+
+  public BatchTransactionService(DBClientFactory dbClientFactory, BatchTransactionDAO transactionDAO,
+      FundService fundService, BudgetService budgetService, LedgerService ledgerService,
+      Set<BatchTransactionServiceInterface> batchTransactionStrategies) {
+    this.dbClientFactory = dbClientFactory;
+    this.transactionDAO = transactionDAO;
+    this.fundService = fundService;
+    this.budgetService = budgetService;
+    this.ledgerService = ledgerService;
+    serviceMap = new EnumMap<>(TransactionType.class);
+    batchTransactionStrategies.forEach(
+      strategy -> serviceMap.put(strategy.getTransactionType(), strategy));
+  }
+
+  public Future<Void> processBatch(Batch batch, RequestContext requestContext) {
+    try {
+      BatchTransactionChecks.sanityChecks(batch);
+    } catch (Exception ex) {
+      logger.error(ex);
```

Comment:
lets add some message to this log, in this it would be easier to understand when reading logs that processing is failed at initial stage of batch processing

Reply:
OK